### PR TITLE
164 Default SMS app for sdk 29+

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

In this PR:
- Adding support for SDK 29+: Using RoleManager to set Gateway as default SMS app.
- Adding some more logs for better debugging in prod. 

Ticket: https://github.com/medic/medic-gateway/issues/164

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
Hi @mrsarm, can you please take a look at this code? Thanks!

File: src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java [hunk 1]
```diff
@@ -25,52 +26,80 @@ public PromptToSetAsDefaultMessageAppActivity() {
...
 		continueToSettings();
 	}
 
-	public void openDefaultMessageAppSettings(View v) {
-		Intent i = new Intent(Intents.ACTION_CHANGE_DEFAULT);
-		i.putExtra(Intents.EXTRA_PACKAGE_NAME, getPackageName());
-		startActivityForResult(i, REQUEST_CHANGE_DEFAULT_MESSAGING_APP);
+	@TargetApi(Build.VERSION_CODES.KITKAT)
+	@SuppressLint({"ObsoleteSdkInt", "InlinedApi"})
+	public void openDefaultMessageAppSettings(View view) {
```
Comment (latin-panda) [inline src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java:63]: I tried to unit test this change, but I couldn't find a nice way of mocking the Build.VERSION.SDK_INT or a way to test which intent got started by the activity, considering our current libs versions. 
Espresso lib didn't have IntentsTestRule or intents.init() 
Please let me know if you know a way to test this :)

Comment (mrsarm) [reply src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java:63]: @ngaruko is adding better testing capabilities and coverage to the medic-android app in https://github.com/medic/medic-android/pull/173, so maybe you can talk with him about it.

**Review by mrsarm — APPROVED** [review]
LGTM !

File: src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java (hunk 1 above)
Comment (latin-panda) [reply src/main/java/medic/gateway/alert/PromptToSetAsDefaultMessageAppActivity.java:63]: @ngaruko If you have any suggestion pls let me know :) For now moving this to AT.

**Pull request merged**